File: minwook/7주차/소수_만들기.java
class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        for(int i=0; i<nums.length-2; i++){
            for(int j=i+1; j<nums.length-1; j++){
                for(int k=j+1; k<nums.length; k++){
                    answer += decimal(nums[i]+nums[j]+nums[k]);
                } 
            }
        }
        return answer;
    }
    
    public int decimal(int n){ // 소수 판별 함수 소수면 1 아니면 0
        for(int i=2; i<n; i++){
            if(n%i==0)
                return 0;
        }
        return 1;
    }
}

File: yeongseong/7week/소수 만들기.js
function isPrime(num){
    // 소수인지 판별 할 함수
    if (num <= 1){ 
        return false;
    }
    for (let i = 2; i <= Math.sqrt(num); i++){
        if (num % i == 0){
            return false
        }
    }
    return true;
}
function solution(nums) {
    let decimal = 0;
    let sum = 0;
    for (let i = 0; i < nums.length; i++){
        for (let j = i + 1; j < nums.length; j++) {
            for (let k = j + 1; k < nums.length; k++){
              sum = nums[i] + nums[j] + nums [k] 
                if (isPrime(sum)){ // 더한 값이 소수인지 확인
                   decimal++
                }
            }
        }
    }
    
    return decimal;
}